### PR TITLE
Record number of checkpoints made in SsdCacheStats

### PR DESCRIPTION
**Pull request by zacw7**
Differential Revision: D56270750

### CODE REVIEW

File: velox/common/caching/SsdFile.cpp [hunk 1]
```diff
@@ -810,7 +811,9 @@ void SsdFile::initializeCheckpoint() {
   }
   bool hasCheckpoint = true;
   std::ifstream state(fileName_ + kCheckpointExtension);
-  if (!state.is_open()) {
+  if (state.is_open()) {
+    ++stats_.numCheckpoints;
```
Comment (xiaoxmeng) [inline velox/common/caching/SsdFile.cpp:815]: I think we shall have a counter inside readCheckpoint: numCheckpointReads?
And add a counter in SsdFile::checkpoint to record the sdd checkpoint write successes?

**Review by xiaoxmeng — COMMENTED** [review]
@zacw7 thanks for the change!

File: velox/common/caching/SsdFile.cpp [hunk 2]
```diff
@@ -920,6 +924,7 @@ void SsdFile::readCheckpoint(std::ifstream& state) {
       entries_[std::move(key)] = run;
     }
   }
+  ++stats_.checkpointsRead;
```
Comment (xiaoxmeng) [inline velox/common/caching/SsdFile.cpp:927]: Put this at the function entry?

**Review by xiaoxmeng — COMMENTED** [review]
@zacw7 Looks good. Can you add unit tests for this? Thanks!

File: velox/common/caching/SsdFile.cpp (hunk 2 above)
Comment (zacw7) [reply velox/common/caching/SsdFile.cpp:927]: Do we want this counter to include failed reads? I think before this point the read can still be interrupted with exception.

**Review by xiaoxmeng — APPROVED** [review]

**Pull request closed without merging**